sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetFilterSuite.scala

```diff
@@ -313,6 +315,36 @@ class ParquetFilterSuite extends QueryTest with ParquetTest with SharedSQLContex
     }
   }
 
+  test("filter pushdown - date") {
+    implicit class IntToDate(int: Int) {
+      def d: Date = new Date(Date.valueOf("2018-03-01").getTime + 24 * 60 * 60 * 1000 * (int - 1))
+    }
+
+    withParquetDataFrame((1 to 4).map(i => Tuple1(i.d))) { implicit df =>
+      checkFilterPredicate('_1.isNull, classOf[Eq[_]], Seq.empty[Row])
+      checkFilterPredicate('_1.isNotNull, classOf[NotEq[_]], (1 to 4).map(i => Row.apply(i.d)))
+
+      checkFilterPredicate('_1 === 1.d, classOf[Eq[_]], 1.d)
```

Comment:
`1.d` is weird, can we name it `1.date`?

Reply:
I agree 1.date is better, but binary is using "1.b", should we keep the same pattern with it?
```
  test("filter pushdown - binary") {
    implicit class IntToBinary(int: Int) {
      def b: Array[Byte] = int.toString.getBytes(StandardCharsets.UTF_8)
    }
```

Reply:
`b` might be OK, but `d` is confusing as it's also a postfix for double.

Reply:
Got it, thanks very much for explanation.